### PR TITLE
refactor: use async block and multihashing interface

### PR DESCRIPTION
**Pull request by dignifiedquire**
Replaces #50

### CODE REVIEW

**Review by haadcode — COMMENTED** [review]
LGTM! This is a much needed fix, let's get it into the release asap!

**Comment by haadcode** [comment]
Erm. The new PR view is confusing. The comment ^ was meant for the fix for `merkeledag.proto` and fs.readFileSync.

File: src/dag-node.js [hunk 1]
```diff
@@ -186,17 +230,25 @@ module.exports = class DAGNode {
...
+        Hash: mh.toB58String(results[1]),
+        Size: results[0]
+      })
+    })
   }
 
   toString () {
-    const hash = mh.toB58String(this.multihash())
-    return `DAGNode <${hash} - data: "${this.data.toString()}", links: ${this.links.length}, size: ${this.size()}>`
+    return `DAGNode <data: "${this.data.toString()}", links: ${this.links.length}>`
```
Comment (diasdavid) [inline src/dag-node.js]: Why change the `toString` this is a breaking change

Comment (daviddias) [inline src/dag-node.js:252]: Why change the toString this is a breaking change

**Review by daviddias — CHANGES_REQUESTED** [review]

File: src/dag-node.js (hunk 1 above)
Comment (dignifiedquire) [reply src/dag-node.js:252]: The alternative is to make `toString` async (to be able to fetch `size` and `hash`), which I found much worse than dropping them from the output

**Comment by dignifiedquire** [comment]
Closing as this module is deprecated now. Ref #62

**Pull request closed without merging**